examples/map/masking_hmi.py

```diff
@@ -0,0 +1,74 @@
+"""
+===============================================
+Masking HMI based on the intensity of AIA Image
+===============================================
+
+In this example we will mask out an HMI Based on the
+intensity values of an AIA image
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+from skimage.measure import label, regionprops
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.data.sample import AIA_171_IMAGE, HMI_LOS_IMAGE
+
+###############################################################################
+# We start with the sample data.
+
+aia = sunpy.map.Map(AIA_171_IMAGE)
+aia = aia.submap(
+    bottom_left=SkyCoord(-250, 200, unit=u.arcsec, frame=aia.coordinate_frame),
+    width=500 * u.arcsec,
+    height=400 * u.arcsec,
+)
+hmi = sunpy.map.Map(HMI_LOS_IMAGE)
+
+###############################################################################
+# Next we call the `reproject_to` funtion to reproject the HMI data to exactly the same grid as the AIA data.
+# Reproject is used to transform one image to the reference frame of another image.
+# See this example :ref:`sphx_glr_generated_gallery_map_transformations_reprojection_align_aia_hmi.py` for more reference.
+
+hmi = hmi.reproject_to(aia.wcs)
+
+###############################################################################
+# Identifying the separate regions below a threshold in the AIA image.
+
+segmented = aia.data < 200
+labeled = label(segmented)
+regions = regionprops(labeled, hmi.data)
+
+###############################################################################
+# Sorting the regions in descending order by size.
+
+regions = sorted(regions, key=lambda r: r.area, reverse=True)
+
+###############################################################################
+# Ploting and numbering the larger regions in the AIA image, According to their size
+# with the region 0 being the largest.
```

Comment:
```suggestion
# Now to plot and label the first 7 regions seen in AIA with the region "0" being the largest.
```

Reply:
Hi @solarkatuwal, you could use the `intensity_mean` property of the `regionprops` function to get the average intensity. As `regions` is our `regionprops` object here, you could use `region[0]['intensity_mean']` to get the average intensity. [Here is the reference,](https://scikit-image.org/docs/0.19.x/api/skimage.measure.html?highlight=region%20props#regionprops) hope this helps :)